partner_address_street3/view/partner_view.xml

```diff
@@ -7,18 +7,27 @@
       <field name="inherit_id" ref="base.view_partner_form"/>
       <field name="arch" type="xml">
         <xpath
-            expr="//form[@string='Partners']/sheet/group/group/div/field[@name='street2']"
+            expr="/form/sheet/group/group/div/field[@name='street2']"
             position="after">
-          <field name="street3"
-                 attrs="{'readonly': [('use_parent_address','=',True)]}"/>
+          <field name="street3" 
+                 placeholder="Street 3..." 
+                 class="o_address_street"
+                 attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
         </xpath>
-
-        <xpath expr="//form[@string='Contact']/sheet/group/div/field[@name='street2']"
+        <xpath expr='//field[@name="child_ids"]' position="attributes">
```

Comment:
This context is not going to be inheritable by multiple modules that adds a partner field.